crm_create_project/wizards/crm_create_project.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2024 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+from odoo import _, fields, models
+
+
+class CrmCreateProject(models.TransientModel):
+    _name = "crm.create.project"
+    _description = "Wizart to create Project from Lead/Opportunity"
+
+    project_name = fields.Char()
+    project_description = fields.Html()
+    lead_id = fields.Many2one("crm.lead")
+
+    def create_project(self):
+        project = (
+            self.env["project.project"]
+            .sudo()
+            .create(self._prepare_create_project_values())
+        )
+        self.lead_id.project_id = project
+        project.message_post(
+            body=_(
+                "Project created from lead/opportunity "
+                "<a href=# data-oe-model=crm.lead data-oe-id=%(lead)d>%(name)s</a>.",
+                lead=self.lead_id,
+                name=self.lead_id.name,
+            )
+        )
```

Comment:
maybe you can create message with origin link
```suggestion
        project.message_post_with_view(
            'mail.message_origin_link', 
            values={'self': project': 'origin' self.lead},
            subtype_id=self.env.ref('mail.mt_note').id, 
            author_id=self.env.user.partner_id.id
        )
```

Reply:
An also check lead_Id as below